rviz_common/include/rviz_common/message_filter_display.hpp

```diff
@@ -134,7 +166,12 @@ class MessageFilterDisplay : public _RosTopicDisplay
         std::bind(
           &MessageFilterDisplay<MessageType>::messageTaken, this,
           std::placeholders::_1));
-      setStatus(properties::StatusProperty::Ok, "Topic", "OK");
+
+      if (!different_qos) {
+        setStatus(properties::StatusProperty::Ok, "Topic", "OK");
+      } else {
+        setStatus(properties::StatusProperty::Warn, "Topic", "Different QoS the publisher");
```

Comment:
Message is not valid English I think: "QoS different from the publisher" would be better maybe

Reply:
Or '1 or more publishers have different QoS settings' to be more accurate with the implementation

Reply:
You are right the second one seems more accurate.

Reply:
Fixed!